### ISSUE TITLE
`if_let_mutex` doesn't trigger if operating on a reference to a mutex

### ISSUE THREAD

**Issue by Dzejkop** [rust-lang/rust-clippy#9193]
### Summary

Lint doesn't trigger when operating on a reference to a mutex, even though such code is functionally the same and incorrect in the same way (causes a deadlock).

### Lint Name

if_let_mutex

### Reproducer

For the following code:

```rust
#![allow(unused)]

use std::sync::Mutex;

pub fn trigger_1() {
    let mutex = Mutex::new(1);

    if let Ok(_) = mutex.lock() {
        do_thing();
    } else {
        mutex.lock();
    };
}

pub fn trigger_2(mutex: &Mutex<i32>) {
    if let Ok(_) = mutex.lock() {
        do_thing();
    } else {
        mutex.lock();
    };
}

pub fn trigger_3(mutex: Mutex<i32>) {
    if let Ok(_) = mutex.lock() {
        do_thing();
    } else {
        mutex.lock();
    };
}

fn do_thing() {}
```

[Playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=7fcc496458d492422969d3dc8e0c6f27)

After running `cargo clippy -- -A clippy::all -D clippy::if_let_mutex`

I expected to see the lint trigger for all the methods, but it only triggers for methods `trigger_1` and `trigger_3`

The output I got:
```
> cargo clippy -- -A clippy::all -D clippy::if_let_mutex
    Checking clippy-lints v0.1.0 (/home/dzejkp/Projects/Personal/clippy-lints)
error: calling `Mutex::lock` inside the scope of another `Mutex::lock` causes a deadlock
  --> src/if_let_mutex.rs:8:5
   |
8  | /     if let Ok(_) = mutex.lock() {
9  | |         do_thing();
10 | |     } else {
11 | |         mutex.lock();
12 | |     };
   | |_____^
   |
   = note: requested on the command line with `-D clippy::if-let-mutex`
   = help: move the lock call outside of the `if let ...` expression
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#if_let_mutex

error: calling `Mutex::lock` inside the scope of another `Mutex::lock` causes a deadlock
  --> src/if_let_mutex.rs:24:5
   |
24 | /     if let Ok(_) = mutex.lock() {
25 | |         do_thing();
26 | |     } else {
27 | |         mutex.lock();
28 | |     };
   | |_____^
   |
   = help: move the lock call outside of the `if let ...` expression
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#if_let_mutex

error: could not compile `clippy-lints` due to 2 previous errors
```

### Version

```text
rustc 1.62.0 (a8314ef7d 2022-06-27)
binary: rustc
commit-hash: a8314ef7d0ec7b75c336af2c9857bfaf43002bfc
commit-date: 2022-06-27
host: x86_64-unknown-linux-gnu
release: 1.62.0
LLVM version: 14.0.5
```
